internal/controller/ols_app_server_assets.go

```diff
@@ -478,6 +479,107 @@ func (r *OLSConfigReconciler) generatePrometheusRule(cr *olsv1alpha1.OLSConfig)
 	return &rule, nil
 }
 
+func (r *OLSConfigReconciler) generateAppServerNetworkPolicy(cr *olsv1alpha1.OLSConfig) (*networkingv1.NetworkPolicy, error) {
+	np := networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OLSAppServerNetworkPolicyName,
+			Namespace: r.Options.Namespace,
+			Labels:    generateAppServerSelectorLabels(),
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: generateAppServerSelectorLabels(),
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{
+				{
+					// allow prometheus to scrape metrics
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{
+									{
+										Key:      "app.kubernetes.io/name",
+										Operator: metav1.LabelSelectorOpIn,
+										Values:   []string{"prometheus"},
+									},
+									{
+										Key:      "prometheus",
+										Operator: metav1.LabelSelectorOpIn,
+										Values:   []string{"k8s"},
+									},
+								},
+							},
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"kubernetes.io/metadata.name": "openshift-monitoring",
+								},
+							},
+						},
+					},
+
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &[]corev1.Protocol{corev1.ProtocolTCP}[0],
+							Port:     &[]intstr.IntOrString{intstr.FromInt(8443)}[0],
+						},
+					},
+				},
+				{
+					// allow the console to access the API
+					From: []networkingv1.NetworkPolicyPeer{
+						{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"app": "console",
+								},
+							},
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"kubernetes.io/metadata.name": "openshift-console",
+								},
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: &[]corev1.Protocol{corev1.ProtocolTCP}[0],
+							Port:     &[]intstr.IntOrString{intstr.FromInt(8443)}[0],
```

Comment:
@raptorsun Should this. be AppServerServiceHTTPSPort - instead of setting as 8443 ?

Reply:
good point. will use the constant 👍